CourseTemplate/CourseTemplate/Program.cs

```diff
@@ -27,13 +33,67 @@
 // Use the CORS policy
 app.UseCors("AllowSpecificOrigin");
 
-var service = new CourseTemplateService();
+app.UseExceptionHandler(errorApp =>
+{
+    errorApp.Run(async context =>
+    {
+        var exceptionHandlerPathFeature =
+            context.Features.Get<Microsoft.AspNetCore.Diagnostics.IExceptionHandlerPathFeature>();
+        context.Response.StatusCode = 500;
 
-app.MapGet("/", () => "Hello World!");
-app.MapGet("/courseTemplate/list", async () => await service.ListAsync());
-app.MapPost("/courseTemplate", async (CourseTemplateDto courseTemplate) =>
+        // Check if the exception is of type ArgumentException
+        if (exceptionHandlerPathFeature?.Error is ArgumentException)
```

Comment:
Lot's of other information can be useful in exception. Expose it for now
Like: StackTrace, Exception.GetType().Name at least